src/Serilog/Parameters/PropertyValueConverter.cs

```diff
@@ -154,17 +155,28 @@ LogEventPropertyValue CreatePropertyValue(object value, Destructuring destructur
                 // Only actual dictionaries are supported, as arbitrary types
                 // can implement multiple IDictionary interfaces and thus introduce
                 // multiple different interpretations.
+                var valueAsList = enumerable.Cast<object>();
```

Comment:
Since 99% of the time the value won't be a dictionary, we should avoid the extra allocation here and just do all of the dictionary handling within the block below.